agent_build/common.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2014-2021 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib as pl
+import re
+import shlex
+import subprocess
+import sys
+import time
+from typing import Union, List
+
+__SOURCE_ROOT__ = pl.Path(__file__).parent.parent
+
+
+def cat_files(file_paths, destination, convert_newlines=False):
+    """Concatenates the contents of the specified files and writes it to a new file at destination.
+
+    @param file_paths: A list of paths for the files that should be read. The concatenating will be done in the same
+        order as the list.
+    @param destination: The path of the file to write the contents to.
+    @param convert_newlines: If True, the final file will use Windows newlines (i.e., CR LF).
+    """
+    dest_fp = open(destination, "w")
+    for file_path in file_paths:
+        in_fp = open(file_path, "r")
+        for line in in_fp:
+            if convert_newlines:
+                line.replace("\n", "\r\n")
+            dest_fp.write(line)
+        in_fp.close()
+    dest_fp.close()
+
+
+def recursively_delete_files_by_name(
+        dir_path: Union[str, pl.Path],
+        *file_names: Union[str, pl.Path]
+):
+    """Deletes any files that are in the current working directory or any of its children whose file names
+    match the specified regular expressions.
+
+    This will recursively examine all children of the current working directory.
+
+    @param file_names: A variable number of strings containing regular expressions that should match the file names of
+        the files that should be deleted.
+    """
+    # Compile the strings into actual regular expression match objects.
+    matchers = []
+    for file_name in file_names:
+        matchers.append(re.compile(str(file_name)))
+
+    # Walk down the current directory.
+    for root, dirs, files in os.walk(dir_path.absolute()):
+        # See if any of the files at this level match any of the matchers.
+        for file_path in files:
+            remove_it = False
+            for matcher in matchers:
+                if matcher.match(file_path):
+                    remove_it = True
```

Comment:
I guess we could just break early for this inner for loop in case we encounter this condition (or simplify it and move deletion directly into this if + then break).